app/controllers/application_controller.rb

```diff
@@ -7,4 +7,131 @@ class ApplicationController < Sinatra::Base
     set :views, 'app/views'
   end
 
```

Comment:
This controller is overloaded. Consider separating your concerns by splitting some of these actions into separate controllers (i.e. a users controller and a tweets controller).